### PR TITLE
DOC: dropdowns in userguide

### PR DESCRIPTION
**Pull request by story645**
## PR Summary

wrapped most of the 2nd level menus in the [user guide](https://matplotlib.org/devdocs/users/index.html) in dropdowns:
image

My reasoning was that this page kind of didn't seem to have enough content to be a site map but had too much to be easily scannable as an index, so I think the drop downs may provide a way to scan for the content folks might want. I mostly kept the titles from each page, except `Explanations->Guide to Matplotlib` 'cause those docs seem like the closest to a narrative user guide.

ETA: also did not touch the ordering except to add a link to the third-party-page 'cause I can never find it and that's honestly why I touched this page in the first place.  

ETA2: [Circle Link](https://output.circle-artifacts.com/output/job/e43c1b9c-04c5-405e-a858-436f127fb441/artifacts/0/doc/build/html/users/index.html)

### CODE REVIEW

**Review by choldgraf — APPROVED** [review]
My 2 cents:

I think that this is a clever way of providing top-level information to help others quickly scan the page without being over-loaded. There's always a trade-off between "hiding things behind a click" and "showing so much that it's overwhelming", and this seems like a reasonable balance between the two. It also somehow feels a bit more polished to me because of the UI elements instead of lists of raw links. I think it's worth trying and seeing what kind of feedback readers give

**Comment by timhoffm** [comment]
I totally agree that the current index is way too much information. Towards that, this is a significant improvement. However, I'm not a fan of using dropdowns here, because you have to click a lot to know where you may want to be going. I'd rather shorten the list of targets. One may or may not still have visual grouping or panels or whatever to make the GUI look nicer.

Shortening: For a number of the sections, we don't need to reference the subsections on the overview page. For example, linking [Installation](https://matplotlib.org/devdocs/users/installing/index.html) is enough. We don't need all the subsection links. They anyway go to the same page, which has the TOC in the sidebar. It's better to just click `Installation` and be on the page and then use the naviation on that page.

**Comment by jklymak** [comment]
I think the User Guide Page is a bit of a misnomer, but that aside, I prefer the information content unfolded as it currently is.  I think it's unattractive, but it's pretty easy to skim for the correct information, whereas folding hides a lot of content.  A multicolumn layout or some such would be fine, but I think the more fundamental issue is the awkward match between the rst docs and the tutorials/gallery; much of what is in the tutorials would more logically be called part of a Users Guide.

**Comment by anntzer** [comment]
> I think the more fundamental issue is the awkward match between the rst docs and the tutorials/gallery; much of what is in the tutorials would more logically be called part of a Users Guide.

FWIW I think that moving the tutorials to use sphinx-gallery was a mistake -- s-g works well for short examples, but the "tutorials" are (should be?) first and foremost text, with some code interspersed, for which the plot directive works just fine (especially with download icons, per https://github.com/matplotlib/matplotlib/pull/24205#issuecomment-1287322046); moving back to plain rst should simplify better organization.

**Comment by tacaswell** [comment]
Either way, I think this is too big to backport.

I am sympathetic to both the pro and con to switching to the foldable things.  Can we start all the folds open?  That way if people just want to skim they can and there is an easy way to hide things that you have already looked to reduce the noise?

**Comment by timhoffm** [comment]
👍 on the structure

Still not on board with the dropdowns. We can't leave vast amount of white space and force the users to expand everything manually:

image

**Comment by jklymak** [comment]
Obviously a taste thing, but I usually resent drilling down webpages to find things.  I think we can use styling or better bullets etc to make the page less overwhelming.

File: doc/users/index.rst [hunk 1]
```diff
@@ -2,40 +2,111 @@
 
 .. redirect-from:: /contents
 
+..
+.. grid:: 3
 
-###########
-Users guide
-###########
+   .. grid-item-card:: Information
```
Comment (timhoffm) [inline doc/users/index.rst:8]: Can you leave the card out and replace it with a regular title? I feel the double nesting of cards and dropdowns is a bit much.

**Review by timhoffm — COMMENTED** [review]
Semi off-topic: This page claims to be the user guide but is actually more a table of contents of the whole page. This is currently by design, but is a mismatch between claimed and actual content.

I think the visual change here is an improvement, but I feel this is not the final solution.

**Comment by timhoffm** [comment]
How about the last one with a `border-top`?

grafik

**Comment by jklymak** [comment]
The problem with the screenshot above is that the title of the subsection has a smaller font (though demi-bold) than topics below them.  That makes it pretty hard to parse and looks like a mistake.

**Comment by jklymak** [comment]
I think there is plenty of room for the heading font size to be larger.  But certainly you should not have the heading font size smaller than its content.

**Comment by timhoffm** [comment]
> Ok so @timhoffm I did figure out the border but probably in the hackiest way and I'm cool w/ commits against this that tweak styles more.

I can have a look, though I’m not a CSS expert either. But likely not before the weekend. This is not necessarily a blocker. We can always fine-tune in a follow-up PR.

File: doc/_static/mpl.css [hunk 2]
```diff
@@ -99,3 +99,170 @@ table.property-table td {
...
+details.sd-dropdown summary {
+    /* padding: 1em; */
+    padding: 0;
+}
+
+.sd-dropdown>.sd-card-body{
+    padding: 0;
+}
+
+:root {
```
Comment (QuLogic) [inline doc/_static/mpl.css:132]: The rest of the file seems to have been duplicated from above.

Comment (QuLogic) [reply doc/_static/mpl.css]: If you expect to use dropdowns elsewhere (or we already have some), then it would be best to add another class on these ones (an ID must be unique, so you'd have to add a bunch here.)

Comment (QuLogic) [reply doc/_static/mpl.css]: It seems simpler to me to just add a single class to every dropdown here and to the CSS selector rather than come up with unique IDs to use for every dropdown and repeat the CSS selector for each one.

File: doc/users/explain/index.rst [hunk 3]
```diff
@@ -2,8 +2,9 @@
 
 .. redirect-from:: /users/explain
 
-Explanations
-------------
+
+API Documentation
```
Comment (tacaswell) [inline doc/users/explain/index.rst:6]: I think this is a confusing change as the difference between "API Reference" and "API Documentation" is very small in my reading.

**Comment by melissawm** [comment]
Oops sorry I had lost this ping! Let me respond below.

**Comment by tacaswell** [comment]
The concrete question is do we try to get this in for 3.7 or push this out to 3.8?
